### DISCUSSION TITLE
Convert JS New() To Brython ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2289] [Q&A] [answered]
Hi,

On the Brython documention page about [JSobject](https://brython.info/static_doc/en/jsobjects.html)

I see that Brython support a way to "translate" the JS new().

I would like to convert this

```js
const doc = new DOMParser().parseFromString(responseText, 'text/html');
```

To Byrhton

I've tried

```python
doc = window.new().DOMParser().parseFromString(s, 'text/html')
'''
output: 
Uncaught (in promise) __dir__:
'''
```
> 

```python
doc = window.DOMParser().parseFromString(s, 'text/html')
'''
output: 
TypeError: DOMParser constructor: 'new' is require
'''
```
Any ideas ?

**Comment by ed2050** [comment]
Hi @BobSquarePants, yes just do window.DOMParser.new () to create the parser object, then call whatever methods you want on it.

I use this helper function in my brython code:
```py
def make_html (text) :
    'Turn a string into html nodes'
    # see https://stackoverflow.com/questions/10585029/parse-an-html-string-with-js

    parser = window.DOMParser.new ()
    html = parser.parseFromString (text, 'text/html')
    return html.body.firstElementChild
```

This works for most cases.  However I recently noticed that it fails if text contains a header element like:
- `<base href=...>` 
- `<style> </style>` (I think)
- or maybe it's `<link rel="stylesheet" href=...>`

Anything that gets put in html.head instead of html.body will fail.  I don't use it for header nodes so I haven't gotten around to fixing it yet.

**Reply by BobSquarePants** [reply]
Thank you @ed2050 , I didn't try your solution yet, but I'll keep it warm when I will need it again.

I still need the `new()` function but something else

I'm binding an event to an element (input) and assign a function
```python

def FooBar(ev):
    print(ev.target.id)

ele.bind('change', FooBar)
```

I would like to launch Foobar once before even a change event occur

Apparently with JS it will look like
```JS
element.dispatchEvent(new Event('change'));
//source: https://stackoverflow.com/questions/35659430/how-do-i-programmatically-trigger-an-input-event-without-jquery
```
I've try several way (based on your reply) but I can't make it work in Brython

```python
ele.dispatchEvent(window.new(Event('change')))
#give error uncaught __dir__
```

**Comment by ed2050** [comment, marked as answer]
This code:

window.new(Event('change'))

makes a new window object.  To create an Event object, do this instead:

window.Event.new('change')


I didn't check the rest of your code, but that will certainly fix one
problem.

**Reply by BobSquarePants** [reply]
Thank you @ed2050 !

I wish FrontEnd was natively Python enabled, and not only JS...
